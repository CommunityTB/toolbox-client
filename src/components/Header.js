import React, { Component } from 'react';
import AutoCompleteWrap from '../components/AutoComplete/AutoCompleteWrap';
import hero from '../assets/images/hero_bg.png';

export default class Header extends Component {

    render() {
        return(
            <header>
                <img src={hero} alt='Community Toolbox Hero' className='hero'></img>
                <AutoCompleteWrap />
            </header>
        )
    }

}

