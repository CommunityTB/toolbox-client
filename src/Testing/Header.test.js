import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from '../components/Header'

describe(`Header component`, () => {
  // Enzyme doesn't support React.createContext so skipping this test
  it.skip('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Header /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a Header by default', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})