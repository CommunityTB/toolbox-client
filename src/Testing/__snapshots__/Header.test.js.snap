// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `
<header>
  <img
    alt="Community Toolbox Hero"
    className="hero"
    src="hero_bg.png"
  />
  <div
    className="middle-align"
  >
    <h1
      className="top-title"
    >
      THE COMMUNITY
    </h1>
    <h1
      className="bottom-title"
    >
      TOOLBOX
    </h1>
    <AutoCompleteWrap />
  </div>
</header>
`;
